lsp/nls/src/cache.rs

```diff
@@ -72,8 +72,8 @@ impl CacheExt for Cache {
             let (_, linearized) =
                 typecheck::type_check_linearize(term, initial_ctxt.clone(), self, host, building)
                     .map_err(|err| vec![Error::TypecheckError(err)])?;
+            lin_registry.insert(file_id, linearized, term);
             self.update_state(file_id, EntryState::Typechecked);
-            lin_registry.map.insert(file_id, linearized);
```

Comment:
I'm pretty sure this doesn't change the behavior. The borrow-checker wanted the re-order, because `term` holds a borrow of `self`, meaning we can't `self.update_state` until we're done with `term`.

There isn't any actual conflict between the two lines, though, because `update_state` only updates the `state` field in the `TermEntry`; it doesn't touch the term.